packages/tools/kadena-cli/src/keys/commands/keysGenFromMnemonic.ts

```diff
@@ -0,0 +1,81 @@
+import type { EncryptedString } from '@kadena/hd-wallet';
+import { kadenaEncrypt, kadenaMnemonicToSeed } from '@kadena/hd-wallet';
+import { kadenaMnemonicToRootKeypair as legacykadenaMnemonicToRootKeypair } from '@kadena/hd-wallet/chainweaver';
+import chalk from 'chalk';
+import type { Command } from 'commander';
+import debug from 'debug';
+import { createCommand } from '../../utils/createCommand.js';
+import { globalOptions } from '../../utils/globalOptions.js';
+import type { IKeysConfig } from '../utils/keySharedKeyGen.js';
+import { generateFromSeed } from '../utils/keySharedKeyGen.js';
+import {
+  displayGeneratedPlainKeys,
+  printStoredPlainKeys,
+} from '../utils/keysDisplay.js';
+import * as storageService from '../utils/storage.js';
+
+import ora from 'ora';
+import { clearCLI } from '../../utils/helpers.js';
+
+export const createGenerateFromMnemonic: (
+  program: Command,
+  version: string,
+) => void = createCommand(
+  'from-mnemonic',
+  'create key(s) from mnemonic phrase',
+  [
+    globalOptions.keyGenFromChoice(),
+    globalOptions.keyMnemonic(),
+    globalOptions.keyPassword(),
+    globalOptions.keyAlias(),
+    globalOptions.keyAmount({ isOptional: true }),
+    globalOptions.legacy({ isOptional: true, disableQuestion: true }),
+  ],
+  async (config) => {
+    clearCLI();
+    try {
+      debug('generate-from-mnemonic:action')({ config });
+
+      const loading = ora('Generating..').start();
```

Comment:
how long does this process usually take? I feel like these operations are usually close to instant, do we need a spinner?

Reply:
This part is using the kadena-crypto package, it’s a lot slower the the slip 10 version. It’s actually takes up to 10 sec for generating 30 tokens. ( since the internals of that package are running some low level commands which is kind of a black box) we have no influence on it